lib/fs/basicfs.go

```diff
@@ -232,7 +259,22 @@ func (f *BasicFilesystem) DirNames(name string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
-
+	if f.encoder != nil {
+		// If we find an encoded filename on disk, but it was encoded by a
+		// encoder other than the current encoder, then we ignore it.
+		// These files must be ignored or the same file would be created
+		// with different encoded names.
+		//
+		// The standard encoder doesn't encode any filenames, so any encoded
+		// filenames found on disk are ignored, as they are probably due to
+		// the user selecting a different encoder, such as fat, and then
+		// having that encoder save files with encoded names. Then, the user
+		// changes the encoder back to standard. If the user wants Syncthing
+		// to "see" these encoded filenames, they need to select an encoder,
+		// such as the fat encoder, that will decode these filenames, or
+		// configure all connected devices to use the passthrough encoder.
+		names = f.encoder.filter(names)
```

Comment:
Doesn't this result in an inconsistent state accessible through a filesystem interface, as in a file might be filtered here, but I can stat, delete, ... it? I'd expect the same treatments for names on all filesystem methods.

Reply:
That's a good question. Presumably, Syncthing wouldn't "see" the files to know to stat or delete them, as the directory list and walk functions ignore these files on disk. But perhaps it found them in the database, from a previous scan. If we want all fs functions to skip over these files (i.e., treat them as non-existent) I will need to add logic to do so. I think this logic could live in fs.rooted(), which can return a not found error. This is a good idea. I'll need to add some tests for this too.

Reply:
I'm about to push these changes to this PR. Unfortunately, I recreated the add/fs-encoder branch off of main, and will force-push these changes. I hope this is acceptable. My apologies if not.